src/blob/index/bloom.rs

```diff
@@ -137,34 +176,87 @@ impl Bloom {
         Ok(Self::from(save))
     }
 
-    pub fn add(&mut self, item: impl AsRef<[u8]>) {
+    pub fn add(&mut self, item: impl AsRef<[u8]>) -> Result<()> {
```

Comment:
I think both `add` and `contains_in_memory` have plenty in common (but `add` lacks 0 check), and should be unified. You can extract function, which accepts function `usize -> bool`, and use `all` in both cases.

Reply:
I have used another solution, please check it

Reply:
LGTM